pyNN/nest/standardmodels/electrodes.py

```diff
@@ -155,3 +155,119 @@ class NoisyCurrentSource(NestCurrentSource, electrodes.NoisyCurrentSource):
     )
     nest_name = 'noise_generator'
 
+
+# Creation of native_electrode_type
+# See Issue 506
+def get_defaults(model_name):
+    valid_types = (int, float, Sequence)
+    defaults = nest.GetDefaults(model_name)
+    variables = defaults.get('recordables', [])
+    ignore = ['archiver_length', 'available', 'Ca', 'capacity', 'elementsize',
+              'frozen', 'instantiations', 'local', 'model', 'needs_prelim_update',
+              'recordables', 'state', 't_spike', 'tau_minus', 'tau_minus_triplet',
+              'thread', 'vp', 'receptor_types', 'events', 'global_id',
+              'element_type', 'type', 'type_id', 'has_connections', 'n_synapses',
+              'thread_local_id', 'node_uses_wfr', 'supports_precise_spikes',
+              'synaptic_elements']
+    default_params = {}
+    default_initial_values = {}
+    for name, value in defaults.items():
+       if name in variables:
+            default_initial_values[name] = value
+    return default_params, default_initial_values
+
+def get_receptor_types(model_name):
+    return list(nest.GetDefaults(model_name).get("receptor_types", ('excitatory', 'inhibitory')))
+
+def native_electrode_type(model_name):
+    """
+    Return a new NativeElectrodeType subclass.
+    """
+    assert isinstance(model_name, str)
+    default_parameters, default_initial_values = get_defaults(model_name)
+    receptor_types = get_receptor_types(model_name)
+    return type(model_name,
+               (NativeElectrodeType,),
+                {'nest_model': model_name,
+                 'default_parameters': default_parameters,
+                 'default_initial_values': default_initial_values,
+                 'injectable': ("V_m" in default_initial_values),
+                 'nest_name': {"on_grid": model_name, "off_grid": model_name}
+                 })
+
+
+class NativeElectrodeType(NestCurrentSource, electrodes.NoisyCurrentSource):
```

Comment:
This should not inherit from `electrodes.NoisyCurrentSource` because
1. it should be usable with *any* NEST current generator, not just `noise_generator` (e.g. `dc_generator`, `ac_generator`, etc.)
2. `electrodes.NoisyCurrentSource` is a base class for *standard* electrodes, not native electrodes.

This class should also go in `nest/electrodes.py`